src/utility.jl

```diff
@@ -127,6 +127,41 @@ function _replace_sqr(str::AbstractString)
     return str
 end
 
+function _replace_POWER(str::AbstractString)
+    pVal = 0.0
+    for i in 1:length(str)-6
+        if str[i:(i+5)] == "POWER("
+            warp = 0
+            pValStart = 0
+            for j in (i+6):length(str)
+                if str[j] == '('
+                    warp += 1
+                elseif str[j] == ')'
+                    warp -= 1
+                elseif str[j] == ','
+                    pValStart = j
+                end
+                if warp == -1
+                    @assert pValStart > 0   # By the time a function is closed, pVal segment must be detected
+                    substart = i+6
+                    subclose = j-1
+                    subStr = str[substart:subclose]
+                    argStr = str[substart:(pValStart-1)]
+                    pVal = parse(str[pValStart+1:subclose])   # Don't allow complicated expression here
+                    str = replace(str, "POWER($subStr)"," ($argStr)^$(pVal)")
+                    subStr = str[substart-5:subclose-2]
+                    subStr = _replace_POWER(subStr)
+                    str = string(str[1:substart-6], subStr, str[subclose-1:end])
+                    break
+                end
+            end
+        end
+        i > length(str)-6 && break
```

Comment:
I am also a little suspicious about this since the `str`'s length is adjusted. I think I should properly dail back the looper `i` here by 4 positions, since from "POWER(" (6) to "()^#" lost 4 characters.

Reply:
True I didn't think about that part. It's actually good that the line gets shorter otherwise we would have different problem :D What do you think about a while loop which just checks whether the length of the remaining part is less than 6 characters?